internal/config/config.go

```diff
@@ -68,6 +68,7 @@ type Parameters struct {
 	VaultTLSConfig           api.TLSConfig
 	Secrets                  []Secret
 	PodInfo                  PodInfo
+	Audience                 string
```

Comment:
Might it make sense to make this a string slice instead so it matches the way its used in the k8s API better?

Reply:
I'm not sure it is a good idea: while for the k8s API, in general, it makes sense to accept multiple audiences, the Vault Kubernetes auth method only accepts [a single audience for each role](https://github.com/hashicorp/vault-plugin-auth-kubernetes/blob/main/path_role.go#L49), which should be already known at this point of the configuration (`audience` and `roleName` are set together, at the same level).

Moreover, including more than one `aud` in the JWT claims increases the risk of misuse of the token, whereas having only Vault as the specified audience provides better security (and, minor, allows to spare 1 line in the SecretProviderClass configuration).

However, if you still see value in changing this to a string slice, of course I can update it, just let me know 😊

Reply:
Good point! Yeah this is a single-purpose token so that makes complete sense.